install

```diff
@@ -93,41 +95,75 @@ install() {
   echo "$FPREFIX Installing Swarmlet"
 
   echo "$PREFIX Installing required packages"
-  $SUDO "$PKG_MANAGER" update -y -qq &>/dev/null
-  $SUDO "$PKG_MANAGER" install -y -qq git apt-transport-https &>/dev/null
-
-  echo "$PREFIX Cloning $SWARMLET_REPO"
-  if [[ -d $SWARMLET_INSTALL_ROOT && $(command -v swarmlet) ]]; then
-    echo "$PREFIX Swarmlet is already installed, updating"
-    # TODO:
-    # pushd $SWARMLET_INSTALL_ROOT >/dev/null
-    # git pull origin "$INSTALL_BRANCH"
-    # popd >/dev/null
-    # swarmlet update
-  else
-    git clone -q $SWARMLET_REPO $SWARMLET_INSTALL_ROOT
+  $SUDO "$PKG_MANAGER" update -y
+  $SUDO "$PKG_MANAGER" install -y git apt-transport-https rsync
+
+  if [[ "$INSTALL_FROM" = "git" ]]; then
+    echo "$PREFIX Cloning $SWARMLET_REPO"
+    if [[ -d $SWARMLET_INSTALL_ROOT && $(command -v swarmlet) ]]; then
+      echo "$PREFIX Swarmlet is already installed, updating"
+      # TODO:
+      # pushd $SWARMLET_INSTALL_ROOT >/dev/null
+      # git pull origin "$INSTALL_BRANCH"
+      # popd >/dev/null
+      # swarmlet update
+    else
+      git clone -q $SWARMLET_REPO $SWARMLET_INSTALL_ROOT
+    fi
+
+    if [[ "$INSTALL_BRANCH" != "master" ]]; then
+      echo "$PREFIX Checking out $INSTALL_BRANCH"
+      pushd $SWARMLET_INSTALL_ROOT >/dev/null
+      git checkout "$INSTALL_BRANCH"
+      popd >/dev/null
+    fi
   fi
 
-  if [[ "$INSTALL_BRANCH" != "master" ]]; then
-    echo "$PREFIX Checking out $INSTALL_BRANCH"
-    pushd $SWARMLET_INSTALL_ROOT >/dev/null
-    git checkout "$INSTALL_BRANCH"
-    popd >/dev/null
+  echo "$PREFIX Initializing Swarmlet"
+
+  # Preserve SWARMLET_INSTALL_ROOT otherwise swarmlet script will fallback on /mnt/gfs/swarmlet to run the installer
+  # Preserve SSH_AUTHORIZED_KEYS as we need it to copy SSH public keys
+  # Preserve SWARMLET_USERNAME and SWARMLET_PASSWORD as we need them to init local registry
+  # Preserve some other environment variables because we need them too :)
+  if [[ "$SUDO" = "sudo" ]]; then
+    SUDO_PARAMS=""
+    PRESERVE_ENV="
+      SWARMLET_INSTALL_ROOT
+      SSH_AUTHORIZED_KEYS
+      SWARMLET_USERNAME
+      SWARMLET_PASSWORD
+      NEW_HOSTNAME
+      ROOT_DOMAIN
+      CREATE_SWAP
+      INSTALL_ZSH
+      INSTALL_MODULES
+      CA_SERVER
+      DEBUG
+    "
+
+    for env_var in ${PRESERVE_ENV}; do
+      SUDO_PARAMS="${SUDO_PARAMS} ${env_var}=${!env_var}"
```

Comment:
```suggestion
      SUDO_PARAMS="$SUDO_PARAMS $ENV_VAR=${!ENV_VAR}"
```

Reply:
Any reason why `${SUDO_PARAMS}` is being used instead of `$SUDO_PARAMS` btw?

Reply:
> Any reason why `${SUDO_PARAMS}` is being used instead of `$SUDO_PARAMS` btw?

https://bertvv.github.io/cheat-sheets/Bash.html

Reply:
We try to follow Shellcheck to maintain consistency. Bash can be messy 😅 

https://github.com/koalaman/shellcheck/wiki/SC2086